### PR TITLE
feat: implement bat spec

### PR DESCRIPTION
**Pull request by threadflow**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Implement bat command spect

**What is the current behavior? (You can also link to an open issue here)**

bat command could auto-complete now

**What is the new behavior (if this is a feature change)?**

The feature before is just a copy of cat that introduce the command briefly. 

Now, the whole auto-complete is worked.

**Additional info:**

### CODE REVIEW

**Comment by threadflow** [comment]
👍

File: src/bat.ts [hunk 1]
```diff
@@ -1,9 +1,312 @@
+import { Script } from "vm";
```
Comment (fedeci) [inline src/bat.ts:1]: Do we need this import?

File: src/bat.ts [hunk 2]
```diff
@@ -1,9 +1,312 @@
...
+                return outLine
+                  .replace(/\:/g, ",")
+                  .split(",")
+                  .map((item) => {
+                    return {
+                      name: item.trim(),
+                      description: descriptionString + " language",
+                    };
+                  });
+              })
```
Comment (fedeci) [inline src/bat.ts:36]:
```suggestion
              out.split("\n").map((outLine) => {
                const langLine = outLine.split(":");
                const descriptionString = langLine[0].trim();
                return outLine
                  .replace(/\:/g, ",")
                  .split(",")
                  .map((item) => {
                    return {
                      name: item.trim(),
                      description: descriptionString + " language",
                    };
                  });
              }
```

**Review by fedeci — CHANGES_REQUESTED** [review]

File: src/bat.ts (hunk 1 above)
Comment (threadflow) [reply src/bat.ts:1]: Sorry, this line copied from template and forget to removed before commit. It can be removed.

File: src/bat.ts (hunk 2 above)
Comment (fedeci) [reply src/bat.ts:36]: Yeah, I missed a return 😁

Comment (fedeci) [reply src/bat.ts:36]: Yeah, I missed a return 😁

Comment (fedeci) [reply src/bat.ts:36]:
```suggestion
              return out.split("\n").map((outLine) => {
                const langLine = outLine.split(":");
                const descriptionString = langLine[0].trim();
                return outLine
                  .replace(/\:/g, ",")
                  .split(",")
                  .map((item) => {
                    return {
                      name: item.trim(),
                      description: descriptionString + " language",
                    };
                  });
              }
```

Comment (threadflow) [reply src/bat.ts:36]: I do test the previous suggested code with adding the 'return'.  It not work.

The code here is used to unpack a 2 dimension array, so the committed code is needed.

Comment (fedeci) [reply src/bat.ts:36]: I see, the I still prefer `Array.flat` method than `.concat` which happens to have some perf issues on some browsers.

Comment (threadflow) [reply src/bat.ts]: I refactored this option with `flat()` and tested. Please check again

**Review by mschrage — APPROVED** [review]

**Pull request merged**